### PR TITLE
`azurerm_kubernetes_cluster`, `azurerm_kubernetes_cluster_node_pool` - support for the `gpu_instance_profile` property

### PR DESCRIPTION
**Pull request by ms-henglu**
```
=== RUN   TestAccKubernetesClusterNodePool_gpuInstanceProfile
=== PAUSE TestAccKubernetesClusterNodePool_gpuInstanceProfile
=== CONT  TestAccKubernetesClusterNodePool_gpuInstanceProfile
--- PASS: TestAccKubernetesClusterNodePool_gpuInstanceProfile (767.65s)

```

### CODE REVIEW

File: internal/services/containers/kubernetes_cluster_node_pool_resource.go [hunk 1]
```diff
@@ -152,6 +152,19 @@ func resourceKubernetesClusterNodePoolSchema() map[string]*pluginsdk.Schema {
 			ForceNew: true,
 		},
 
+		"gpu_instance_profile": {
```
Comment (stephybun) [inline internal/services/containers/kubernetes_cluster_node_pool_resource.go:155]: Profile is redundant here, can we rename this to 
```suggestion
		"gpu_instance": {
```

**Review by stephybun — CHANGES_REQUESTED** [review]

File: internal/services/containers/kubernetes_cluster_node_pool_resource.go (hunk 1 above)
Comment (ms-henglu) [reply internal/services/containers/kubernetes_cluster_node_pool_resource.go]: Got it! I've updated this PR.

File: internal/services/containers/kubernetes_cluster_node_pool_resource_test.go [hunk 2]
```diff
@@ -2706,3 +2721,38 @@ resource "azurerm_kubernetes_cluster_node_pool" "test" {
...
+  }
+  identity {
+    type = "SystemAssigned"
+  }
+}
+
+resource "azurerm_kubernetes_cluster_node_pool" "test" {
+  name                  = "internal"
+  kubernetes_cluster_id = azurerm_kubernetes_cluster.test.id
+  vm_size               = "Standard_ND96asr_v4"
```
Comment (stephybun) [inline internal/services/containers/kubernetes_cluster_node_pool_resource_test.go:2754]: Can we please use a smaller, less expensive vm. We don't need to spin up a vm with 96 vCPUs and 900 GB memory to test this?

**Review by stephybun — CHANGES_REQUESTED** [review]
Thanks for making the rename, one last comment on the vm we're using the test which I missed in the initial review, otherwise this is looking good.

File: internal/services/containers/kubernetes_cluster_node_pool_resource_test.go (hunk 2 above)
Comment (ms-henglu) [reply internal/services/containers/kubernetes_cluster_node_pool_resource_test.go]: Got it! I've changed the vmsize.

**Review by stephybun — APPROVED** [review]
Thanks @ms-henglu LGTM 👍

**Pull request merged**